docs/sources/next/using-k6/modules.md

```diff
@@ -380,11 +205,42 @@ $ docker run --rm -v /home/k6/example/src:/src -i grafana/k6 run /src/index.js
 {{< /code >}}
 
 Note that on Windows, you also need to make sure that your drive in question, say `C:\`,
-has been marked for sharing in the Docker settings:
+has been marked for sharing in the Docker Desktop settings.
+
+### Extension modules
+
+The official [Grafana k6 Docker image](https://hub.docker.com/r/grafana/k6) includes the k6 release binary but lacks additional k6 extensions. Therefore, using the official Docker container to run a k6 test that requires an extension will fail.
 
-![Running k6 in docker on Windows](/media/docs/k6-oss/running-k6-in-docker-on-windows.png)
+To run k6 with any extension in Docker, you must create a custom Docker image that includes the k6 binary with the necessary extensions. To learn building a Docker image with k6 and extensions, refer to the [xk6 command](https://grafana.com/docs/k6/<K6_VERSION>/extensions/build-k6-binary-using-go#breaking-down-the-xk6-command) and this [Dockerfile](https://github.com/grafana/xk6-output-prometheus-remote/blob/main/Dockerfile) example.
+
+```bash
+# Stage 1 - download xk6 and desired extensions, then compile a new binary.
+FROM golang:1.20-alpine3.18 as builder
```

Comment:
Not certain we want to have this in docs and then forget to update them all the time.

Maybe linking to xk6 repo and it docs will be better?

We do bump go version all the time. Go 1.20 is no logner supported from today as 1.22 got released and we already mostly build with 1.21. 

Similarly alpine is on version 3.19

Reply:
The minimal Dockerfile to produce an image with custom `k6` build would be

```Dockerfile
FROM grafana/xk6:latest

RUN GCO_ENABLED=0 xk6 build \
    --with github.com/grafana/xk6-output-prometheus-remote@v0.3.1

ENTRYPOINT ["./k6"]
```

If necessary, one could implement a multistage build and copy the `xk6 build` artefact, but I'd leave it out to users.

Reply:
A related thought: is it possible to source the files from repo to the doc? I.e. to have a full file inserted when docs are published. IMHO, in this case, it is better to have a full listing instead of a link to repo and it would also result in the latest instruction set, without the need to constantly update it.
I suppose cc @heitortsergent ?

Reply:
https://github.com/grafana/k6-docs/pull/1489/commits/95fb2e43a1cb2d81d37285d54a80c093594fd320 adds @andrewslotin instructions.

These instructions are very easy to follow ❤️. They show quickly the `xk6 build` command, which is the core topic we want users to learn for their particular case. 


>  is it possible to source the files from repo to the doc? 

@heitortsergent @yorugac, please, let's discuss this separately. 


I am resolving this conversion. Please, reopen if it needs further discussion.

Reply:
I opened https://github.com/grafana/k6-docs/issues/1508 so we can discuss this separately. :)

I know we have the ability to import files from different repos, but I'm not sure about the details.